theseus/optimizer/nonlinear/nonlinear_optimizer.py

```diff
@@ -460,3 +467,22 @@ def _optimize_impl(
     @abc.abstractmethod
     def compute_delta(self, **kwargs) -> torch.Tensor:
         pass
+
+    # Resets any internal state needed by the optimizer for a new optimization
+    # problem. Optimizer loop will pass all optimizer kwargs to this method.
+    def reset(self, **kwargs) -> None:
```

Comment:
Should this be an abstract method?

Reply:
I prefer not, since I imagine there might be solvers where this is not needed (this is indeed the current case for `GaussNewton`), then we are forcing the class to unnecessarily implement a stub just to comply with the base class.